include/ccmath/math/nearest/rint.hpp

```diff
@@ -8,7 +8,117 @@
 
 #pragma once
 
+#include <ccmath/internal/support/fenv/fenv_support.hpp>
+#include <ccmath/internal/support/fenv/rounding_mode.hpp>
+#include <ccmath/internal/support/fp/directional_rounding_utils.hpp>
+#include <ccmath/math/compare/isinf.hpp>
+#include <ccmath/math/compare/isnan.hpp>
+
 namespace ccm
 {
+	template <class T, std::enable_if_t<std::is_floating_point_v<T>, bool> = true>
+	constexpr T rint(T num) noexcept
+	{
+		if (ccm::isinf(num) || ccm::isnan(num)) { ccm::support::fenv::raise_except_if_required(FE_INVALID); }
+		constexpr auto rounding_mode{ccm::support::fenv::get_rounding_mode()};
+		return ccm::support::fp::directional_round(num, rounding_mode);
+	}
+
+	template <class Integer, std::enable_if_t<std::is_integral_v<Integer>, bool> = true>
+	constexpr double rint(Integer num)
```

Comment:
Add noexcept? Also need to document each function.

Reply:
Following the defined API of the standard there should not be a `noexcept` for ccm::rint. This may change later in the future but for the time being aim to mimic the API pretty closely.

Reply:
You can also use this as a reference:

https://en.cppreference.com/w/cpp/numeric/math/rint

Reply:
@Rinzii Sounds good. I removed the noexcept for that method!

I think it is ready for a review!